platforms/nuttx/NuttX/Make.defs.in

```diff
@@ -36,6 +36,18 @@
 include $(TOPDIR)/.config
 include $(TOPDIR)/tools/Config.mk
 
+CINCPATH := $(shell $(INCDIR) -s "$(CC)" $(TOPDIR)$(DELIM)include)
+CXXINCPATH := $(shell $(INCDIR) -s "$(CC)" $(TOPDIR)$(DELIM)include$(DELIM)cxx)
+
+ARCHINCLUDES += $(CINCPATH)
+ARCHXXINCLUDES += $(CINCPATH) $(CXXINCPATH)
+
+ifeq ($(CONFIG_CYGWIN_WINTOOL),y)
```

Comment:
The include paths are all broken on Cygwin with this change. `ifeq ($(CONFIG_CYGWIN_WINTOOL),y)` doesn't even detect Cygwin.

We'll probably need something like:
```
ifneq (, $(findstring CYGWIN, $(shell uname)))
  CONFIG_CYGWIN_WINTOOL = y
```
I started testing but don't have it working yet.

Reply:
I'm still looking for the correct way to deal with this. I have the detection and the TOPDIR conversion but still have to figure out the BOARD_DIR again. I feel it's all deja vu but forgot how it worked. I think last time I looked into it was 2018 https://github.com/PX4/Firmware/pull/8737